specification/cognitiveservices/data-plane/AnomalyDetector/stable/v1.1/cadl/multivariate/models.cadl

```diff
@@ -0,0 +1,370 @@
+import "@cadl-lang/rest";
+import "@azure-tools/cadl-azure-core";
+
+using Cadl.Http;
+using Cadl.Rest;
+using Azure.Core;
+
+namespace AnomalyDetector.Multivariate;
+
+enum FillNAMethodKnownValues {
+  "Previous",
+  "Subsequent",
+  "Linear",
+  "Zero",
+  "Fixed",
+}
+
+@doc("An optional field, indicating how missing values will be filled. One of Previous, Subsequent, Linear, Zero, Fixed.")
+@knownValues(FillNAMethodKnownValues)
+model FillNAMethod is string;
+
+enum DetectionStatus {
+  Created: "CREATED",
+  Running: "RUNNING",
+  Ready: "READY",
+  Failed: "FAILED",
+}
+
+@doc("Data schema of input data source: OneTable or MultiTable. The default DataSchema is OneTable.")
+enum DataSchema {
+  @doc("OneTable means that your input data are all in one CSV file, which contains one 'timestamp' column and several variable columns. The default DataSchema is OneTable.")
+  "OneTable",
+  @doc("MultiTable means that your input data are separated in multiple CSV files, in each file containing one 'timestamp' column and one 'variable' column, and the CSV file name should indicate the name of the variable. The default DataSchema is OneTable.")
+  "MultiTable",
+}
+
+enum AlignMode {
+  "Inner",
+  "Outer",
+}
+
+enum ModelStatus {
+  Created: "CREATED",
+  Running: "RUNNING",
+  Ready: "READY",
+  Failed: "FAILED",
+}
+
+@doc("Detection results for the given resultId.")
+@resource("multivariate/detect-batch")
+model DetectionResult {
+  @doc("Result identifier, which is used to fetch the results of an inference call.")
+  @key
+  @format("uuid")
+  resultId: string;
+
+  @doc("Multivariate anomaly detection status.")
+  summary: DetectionResultSummary;
+
+  @doc("Detection result for each timestamp.")
+  results: AnomalyState[];
+}
+
+@doc("Multivariate anomaly detection status.")
+model DetectionResultSummary {
+  @doc("Status of detection results. One of CREATED, RUNNING, READY, and FAILED.")
+  status: DetectionStatus;
+
+  @doc("Error message when detection is failed.")
+  errors?: ErrorResponse[];
+  variableStates?: VariableState[];
+
+  @doc("""
+Detection request for batch inference. This is an asynchronous inference which
+will need another API to get detection results.
+""")
+  setupInfo: DetectionRequest;
+}
+
+@error
+model ErrorResponse {
+
+  @doc("The error code.")
+  code: string;
+
+  @doc("The message explaining the error reported by the service.")
+  message: string;
+}
```

Comment:
Please see if it can be Error from cadl-azure-core https://github.com/Azure/cadl-azure/blob/main/packages/cadl-azure-core/lib/foundations.cadl#L88

Reply:
Also there is comment in SDK that the properties using this "ErrorResponse" likely be read-only (user likely should not set an error in input...).